src/codeManager.ts

```diff
@@ -451,21 +451,24 @@ export class CodeManager implements vscode.Disposable {
         }
         const showExecutionMessage = this._config.get<boolean>("showExecutionMessage");
         this._outputChannel.show(this._config.get<boolean>("preserveFocus"));
-        const exec = require("child_process").exec;
+        const spawn = require("child_process").spawn;
         const command = this.getFinalCommandToRunCodeFile(executor, appendFile);
         if (showExecutionMessage) {
             this._outputChannel.appendLine("[Running] " + command);
         }
         this.sendRunEvent(executor, false);
         const startTime = new Date();
-        this._process = exec(command, { cwd: this._cwd });
+        this._process = spawn(command, [], { cwd: this._cwd, shell: true });
+
+        const TextDecoder = require("util").TextDecoder;
+        const decoder = new TextDecoder();
 
         this._process.stdout.on("data", (data) => {
-            this._outputChannel.append(data);
+            this._outputChannel.append(decoder.decode(data));

```

Comment:
What's the difference between `decoder.decode(data)` and `data.toString()`? Others look good to me!

Reply:
Calling `toString()` on `data` (a `uint8array`) results in a string representation of the int array, not the actual decoded value.

Also see [this](https://stackoverflow.com/a/36949791/2789451)

Reply:
Seems like the official doc also use `data.toString()` :https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options
And I have tried `data.toString()`, the output is the same `decoder.decode(data)`. Just want to know if there is any case needed to use `decoder.decode(data)`, since `data.toString()` would be simple.

Reply:
You're right, this was a couple of months ago so I don't remember exactly why I used TextDecoder. I have made the change.